src/vs/workbench/contrib/notebook/browser/notebookEditorWidget.ts

```diff
@@ -1692,8 +1692,11 @@ export class NotebookEditorWidget extends Disposable implements INotebookEditorD
 
 	private _restoreSelectedKernel(viewState: INotebookEditorViewState | undefined): void {
```

Comment:
Root cause of the problem was that when creating an interactive window, a 'preselected' kernel can be passed. Depending upon the timing for when the viewstate loads, it can override this preselected kernel.

Reply:
I believe the timing for the viewstate was pushed back because of something else recently and it's why this is suddenly consistently happening. 

The original issue was intermittent: https://github.com/microsoft/vscode-jupyter/issues/10613